tests/serverpod_test_server/test_integration/test_tools/authentication_validation_test.dart

```diff
@@ -87,6 +90,199 @@ void main() {
             .toList();
         expect(result, [0, 1, 2]);
       });
+
+      group('when connected to an authenticated streaming method', () {
+        late Completer<dynamic> streamClosedCompleter;
+        late Completer<int> valueReceivedCompleter;
+        late StreamController<int> inStream;
+
+        var authenticatedUserId = 1;
+
+        setUp(() async {
+          streamClosedCompleter = Completer<dynamic>();
+          inStream = StreamController<int>();
+          Stream<int> outStream;
+          var authenticatedSession = session.copyWith(
+            authentication:
+                AuthenticationOverride.authenticationInfo(1, {Scope('user')}),
+          );
+          outStream = endpoints.authenticatedTestTools.intEchoStream(
+            authenticatedSession,
+            inStream.stream,
+          );
+
+          valueReceivedCompleter = Completer<int>();
+          outStream.listen((event) {
+            if (valueReceivedCompleter.isCompleted) {
+              return;
+            }
+            valueReceivedCompleter.complete(event);
+          }, onError: (e) {
+            streamClosedCompleter.complete(e);
+          });
+
+          inStream.add(1);
+          // Validate that the stream works
+          await valueReceivedCompleter.future;
+          assert(valueReceivedCompleter.isCompleted);
+        });
+
+        test(
+            'and the authenticated user is revoked then stream is closed with ConnectionClosedException.',
+            () async {
+          await expectLater(
+            session.messages.authenticationRevoked(
+              authenticatedUserId,
+              RevokedAuthenticationUser(),
+            ),
+            completion(true),
+          );
+
+          await expectLater(
+            streamClosedCompleter.future.timeout(
+              Duration(seconds: 5),
+            ),
```

Comment:
The timeout is very helpful for TDD. If this completer doesn't complete (which was the case before I added support for this) then you have to wait 30 seconds for the global test timeout. In this case, it's much better to timeout quicker where you expect that this can happen. It's also clear what line in the code that did timeout. I found this very helpful and I think future developers will find it helpful as well if bugs occur in this code 🐛

Reply:
I agree on the purpose of the timeout, we should have it 👍

But from what I remembered when writing the tests for the websocket streams the output didn't really point me to the right place in the code if it failed. For example any reason provided won't show with the test.

Let's try it out together, I might be wrong!

Reply:
Sorry, I misunderstood you slightly! Then I see. For now I will simply remove the `Timeout` since it's only in two places. It can be added back if one has to debug these tests.